src/main/java/com/google/devtools/build/lib/actions/ResourceSet.java

```diff
@@ -43,12 +45,24 @@
   /** The number of CPUs, or fractions thereof. */
   private final double cpuUsage;
 
+  /** Map of extra resources mapping name of the resource to a value. */
+  private final ImmutableMap<String, Float> extraResourceUsage;
+
   /** The number of local tests. */
   private final int localTestCount;
 
   private ResourceSet(double memoryMb, double cpuUsage, int localTestCount) {
+    this(memoryMb, cpuUsage, null, localTestCount);
+  }
+
+  private ResourceSet(double memoryMb, double cpuUsage, ImmutableMap<String, Float> extraResourceUsage, int localTestCount) {
     this.memoryMb = memoryMb;
     this.cpuUsage = cpuUsage;
+    if (extraResourceUsage == null) {
+        this.extraResourceUsage = ImmutableMap.of();
+    } else {
+        this.extraResourceUsage = extraResourceUsage;
+    }
```

Comment:
For me this checks on null looks overcomplicated and it's better to avoid nulls in code. So I suggest mark  `extraResourceUsage` in second constructor as `@NonNull` and use `ImmutableMap.of()` in the first constructor.
Also it's better to use 3 param constructor on other code where it's possible.

Reply:
That seems to work with a couple of caveats:
- I had to add new dependencies in 
- Using `@NonNull` decorator in `ResourceSet.create` resulted in `//src/test/shell/bazel:bazel_bootstrap_distfile_tar_test` test failures with `error: No generator for: (@io.reactivex.rxjava3.annotations.NonNull :: com.google.common.collect.ImmutableMap<java.lang.String,java.lang.Float>)` (see CI job [here](https://buildkite.com/bazel/bazel-bazel-github-presubmit/builds/9343)). I removed `@NonNull` from that function and only kept in the constructor - do you think that's OK?